controllers/core/openstackversion_controller.go

```diff
@@ -0,0 +1,295 @@
+	defer func() {
+		// update the Ready condition based on the sub conditions
+		if instance.Status.Conditions.AllSubConditionIsTrue() {
+			instance.Status.Conditions.MarkTrue(
+				condition.ReadyCondition, condition.ReadyMessage)
+		} else {
+			// something is not ready so reset the Ready condition
+			instance.Status.Conditions.MarkUnknown(
+				condition.ReadyCondition, condition.InitReason, condition.ReadyInitMessage)
+			// and recalculate it based on the state of the rest of the conditions
+			instance.Status.Conditions.Set(
+				instance.Status.Conditions.Mirror(condition.ReadyCondition))
+		}
+
+		err := versionHelper.PatchInstance(ctx, instance)
+		if err != nil {
+			_err = err
+			return
+		}
+	}()
+
+	// greenfield deployment
+	cl := condition.CreateList(
+		condition.UnknownCondition(corev1beta1.OpenStackVersionInitialized, condition.InitReason, string(corev1beta1.OpenStackVersionInitializedInitMessage)),
+	)
+	// no minor update conditions unless we have a deployed version
+	if instance.Status.DeployedVersion != nil && instance.Spec.TargetVersion != *instance.Status.DeployedVersion {
+		cl = append(cl, *condition.UnknownCondition(corev1beta1.OpenStackVersionMinorUpdateOVNControlplane, condition.InitReason, string(corev1beta1.OpenStackVersionMinorUpdateInitMessage)),
+			*condition.UnknownCondition(corev1beta1.OpenStackVersionMinorUpdateControlplane, condition.InitReason, string(corev1beta1.OpenStackVersionMinorUpdateInitMessage)))
+		// fixme add dataplane conditions here
+		//condition.UnknownCondition(corev1beta1.OpenStackVersionMinorUpdateOVNDataplane, condition.InitReason, string(corev1beta1.OpenStackVersionMinorUpdateInitMessage)),
+		//condition.UnknownCondition(corev1beta1.OpenStackVersionMinorUpdateDataplane, condition.InitReason, string(corev1beta1.OpenStackVersionMinorUpdateInitMessage)))
+	}
+	instance.Status.Conditions.Init(&cl)
+	if isNewInstance {
+		// Register overall status immediately to have an early feedback e.g. in the cli
+		return ctrl.Result{}, nil
+	}
+
+	instance.Status.Conditions.Set(condition.FalseCondition(
+		corev1beta1.OpenStackVersionInitialized,
+		condition.RequestedReason,
+		condition.SeverityInfo,
+		corev1beta1.OpenStackVersionInitializedReadyRunningMessage))
+
+	instance.Status.AvailableVersion = &envAvailableVersion
+	defaults := openstack.InitializeOpenStackVersionImageDefaults(ctx, envContainerImages)
+	// store the defaults for the currently available version
+	if instance.Status.ContainerImageVersionDefaults == nil {
+		instance.Status.ContainerImageVersionDefaults = make(map[string]*corev1beta1.ContainerDefaults)
+	}
+	instance.Status.ContainerImageVersionDefaults[envAvailableVersion] = defaults
+
+	// calculate the container images for the target version
+	val, ok := instance.Status.ContainerImageVersionDefaults[instance.Spec.TargetVersion]
+	if !ok {
+		Log.Info("Target version not found in defaults", "targetVersion", instance.Spec.TargetVersion)
```

Comment:
Should we set some condition here to make this state visible? Is this an error state?

Reply:
lets address this in a follow up.

Reply:
OK